services/flows-operator/src/FlowOperator.js

```diff
@@ -111,19 +117,29 @@ class FlowOperator {
                         await channel.deleteExchange(exchange);
                     }
                 }
-                const queues =  await this._queueCreator.makeQueuesForTheTask(flowModel);
-                for (let node of flowModel.nodes) {
+                const queues =  await this._queueCreator.makeQueuesForTheFlow(flow);
+                for (let node of flow.nodes) {
                     this._logger.trace({flow: flow.metadata.name, node: node.id}, 'Going to create flow node');
-                    await this._deployNode(flowModel, node, queues);
+                    await this._deployNode(flow, node, queues);
                 }
             } else {
                 this._logger.trace({name: flow.metadata.name}, 'Nothing changed. Ensure nodes and queues exists');
+
+                if (!jobsIndex[flowId]) {
```

Comment:
I'd rather add code that ensures secret above this: just  before the logic that decides should we totally redeploy the whole task due to changes.
Because in that case we ensure secret even if task should be redeployed